### PR TITLE
Do it for me purchase flow skeleton

### PR DESCRIPTION
**Pull request by ddc22**
#### Changes proposed in this Pull Request

* The initial purchase flow implementation for the DIFM purchase flow
* Flow includes
     *  Landing page `/marketing/do-it-for-me/landing/${siteSlug}`
     *  Site Info page `/marketing/do-it-for-me/site-info/${siteSlug}`
* On submit of the site info page the premium plan and the difm lite product is added to the basket 
* Strings have not been queued for translation because the design/content is yet to be finalised and the initial release will mostly be to a small pilot of users.
* The feature is not implemented behind a feature flag because we will be piloting to only a few number of users.

Screenshots:
|      |   |
| ----------- | ----------- |
| image |  image       |
| image         |        |

#### Testing instructions

- Go to`/marketing/do-it-for-me/landing/${siteSlug}` 
- Click through the call to action buttons
- Make sure both the difm and premium plan are available at checkout.

### CODE REVIEW

File: client/my-sites/marketing/do-it-for-me/landing.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { Button, Card } from '@automattic/components';
+import styled from '@emotion/styled';
+import page from 'page';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import CardHeading from 'calypso/components/card-heading';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+
+const VeticalsGrid = styled.div`
```
Comment (dzver) [inline client/my-sites/marketing/do-it-for-me/landing.tsx]: Did you mean VerticalsGrid?

**Review by dzver — APPROVED** [review]
LGTM

File: client/my-sites/marketing/index.js [hunk 2]
```diff
@@ -26,6 +29,8 @@ export default function () {
 		'/sharing',
 		'/sharing/buttons',
 		'/marketing/business-tools',
+		'/marketing/do-it-for-me/landing',
```
Comment (aneeshd16) [inline client/my-sites/marketing/index.js]: Maybe we can wrap these in a `config.isEnabled( 'difm-lite-pilot' )` as well?

**Review by aneeshd16 — COMMENTED** [review]
Left a minor comment. Everything else LGTM! 👍

File: client/my-sites/marketing/index.js (hunk 2 above)
Comment (ddc22) [reply client/my-sites/marketing/index.js]: Thanks! nice catch :)

**Pull request merged**